magento/pwa-studio
# [fix] offline cached search and category pages

## Description

Category and Search pages were not rendering cached data when offline. This is because we rendered errors even when we had data to render otherwise.

A good example of the "correct" implementation for offline functionality would be the [Product Root Component](https://github.com/magento/pwa-studio/blob/develop/packages/venia-ui/lib/RootComponents/Product/product.js#L28) which only rendered the error when there was no product data to render.

I applied this logic to the category and search pages.

## Related Issue





Closes [PWA-1237](https://jira.corp.magento.com/browse/PWA-1237).

## Acceptance



### Verification Stakeholders

### Specification



### Verification Steps


1. Check out this PR.
2. In your `.env`, set `DEV_SERVER_SERVICE_WORKER_ENABLED=true`.
3. Run `yarn watch:all`.
4. Load the Home Page, then a Category Page, then a Product Page. Then search for that product and remember the term you use.
5. Switch to offline mode. Click the home page. It should render.
6. Now click the Category. It should render.
7. Now click the Product. It should render.
8. Now search using the term from the prior step. It should render the search results.
9. 
## Screenshots / Screen Captures (if appropriate)

## Checklist

* I have added tests to cover my changes, if necessary.
* I have added translations for new strings, if necessary.
* I have updated the documentation accordingly, if necessary.
